idea_town/frontend/static-src/app/collections/experiments.js

```diff
@@ -13,6 +15,22 @@ export default Collection.extend({
   // updated yet? TODO: investigate
   ajaxConfig: { headers: { 'Accept': 'application/json' }},
 
+  initialize() {
+    // TODO:(DJ) This is not being run on the experiment-page.
+    // Also unsure whether it is even being set on the /home page.
+    // It may be that the model is being wiped out after we set it.
+    // Need to ensure we can actually set _client-side-only_
+    // properties. Or else just add it to the dbmodel.
```

Comment:
Yeah, about this todo - I definitely agree, it seems like life would be a lot simpler if the server had all the necessary information ahead of time. Since the addon sends lifecycle events to the server, the server should almost always have that info already.

I think you'll probably see weird race conditions if drawing the page depends on addon-frontend communication.